src/IIIFPresentation/Repository/Helpers/ManifestRetrieval.cs

```diff
@@ -0,0 +1,80 @@
+﻿using Core.Exceptions;
+using Microsoft.EntityFrameworkCore;
+using Models.Database.Collections;
+using Models.Database.General;
+
+namespace Repository.Helpers;
+
+public class ManifestRetrieval
+{
+    /// <summary>
+    ///     For given manifest, return the full hierarchical path for it, delimited by `/`
+    ///     This will consist of the slug of searched for given manifest and all it's parents.
+    /// </summary>
+    /// <param name="manifest">Manifest to retrieve full path for</param>
+    /// <param name="dbContext">Current db context</param>
+    /// <returns>Delimited path</returns>
+    /// <exception cref="PresentationException">Thrown if a circular dependency is expected</exception>
+    /// <remarks>Note that both 'root' level items and not-found items will return empty string</remarks>
+    public static Task<string> RetrieveFullPathForManifest(Manifest manifest, PresentationContext dbContext,
+        CancellationToken cancellationToken = default) =>
+        RetrieveFullPathForManifest(manifest.Id, manifest.CustomerId, dbContext, cancellationToken);
+
+    public static async Task<string> RetrieveFullPathForManifest(string manifestId, int customerId,
```

Comment:
This method is similar to `RetrieveFullPathForCollection` - is it worth looking to see if they can be combined?